lib/key.rb

```diff
@@ -0,0 +1,13 @@
+class Key
+
+  attr_reader :digits
+
+  def initialize(key = '')
+    @digits = (key == '' ? (Array.new(5) { rand(1...9) }).join : key)
```

Comment:
generates an array with 5 random digits and joins them into a string (or takes the key passed as an argument)